modules/persistent-storage-csi-migration-automatic-ga.adoc

```diff
@@ -0,0 +1,19 @@
+// Module included in the following assemblies:
+//
+// * storage/container_storage_interface/persistent-storage-csi-migration.adoc
+
+:_content-type: CONCEPT
+[id="persistent-storage-csi-migration-automatic-ga_{context}"]
+= Automatic migration of in-tree volumes to CSI
+
+{product-title} supports automatic and seamless migration for the following in-tree volume types to their Container Storage Interface (CSI) driver counterpart:
+
+* Azure Disk
+
+* OpenStack Cinder
+
+CSI migration for these volume types is considered generally available (GA), and requires no manual intervention.
+
+After upgrading from {product-title} version 4.10, and earlier, to 4.11, or later, the new default storage class is set to the CSI storage class. If you are installing a new cluster, the CSI storage class is the default, and all volumes provisioned using this storage class are CSI persistent volumes (PVs). The in-tree storage class remains, and might be necessary for certain features, such as volume expansion to work for existing in-tree PVs.
+
+However, for migrated volumes, the default storage class is not changed and remains the in-tree storage class, or whichever storage class was the default before the upgrade. While storage class referencing to the in-tree storage plug-in will continue working, we recommend that you switch the default storage class to the CSI storage class.
```

Comment:
I am a bit confused here - is there a difference between the upgrade scenario in the prior paragraph and the upgrade scenario in this paragraph?

Reply:
The upgrade scenario is the same. I think it's a bit confusing because first paragraph mentions upgrades and fresh installation and the second paragraph then refers again to upgrade scenario.

I'd reword this a bit to make a clear distinction between install and upgrade. And maybe add more details on how upgrade handles the storage classes. Because there's more scenarios that can happen while upgrading: users might already have some storage class that has a same name and would clash with our CSI one, or maybe they have some other default already set. So the fist sentence "After upgrading, the new default storage class is set..." is misleading because in fact it depends on pre-upgrade configuration.